lib/spack/spack/util/windows_registry.py

```diff
@@ -252,11 +299,11 @@ def get_matching_subkeys(self, subkey_name):
 
     def get_values(self):
         if not self._valid_reg_check():
-            raise RegistryError("Cannot query values from invalid key %s" % self.key)
+            raise RegistryError(f"Cannot query values from invalid key {self.key}")
         with self.invalid_reg_ref_error_handler():
             return self.reg.values
 
-    def _traverse_subkeys(self, stop_condition, collect_all_matching=False):
+    def _traverse_subkeys(self, stop_condition, collect_all_matching=False, depth_search=True):
```

Comment:
Can you update docstring with new arg?

Also, did we agree elsewhere that this concept should generally be represented by the term "recursive" (`find_keys` also introduces `depth`) or do you think this should have a distinct term?

Reply:
I think we've gone back and forth a few times on this. I think for user facing (non `_` prefixed) methods, we should use recursive as that's the idiom. 
For internal methods, I think it would be a bit off if they had a different arg name, so I'm ok with everything being `recursive`.